cpp/src/arrow/array/builder_nested.cc

```diff
@@ -216,12 +220,22 @@ Status FixedSizeListBuilder::AppendNull() {
   return value_builder_->AppendNulls(list_size_);
 }
 
+void FixedSizeListBuilder::UnsafeAppendNull() {
+   UnsafeAppendToBitmap(false);
+   (void) value_builder_->AppendNulls(list_size_);
```

Comment:
We don't want to ignore error returns, so instead we should call a hypothetical `UnsafeAppendNulls(list_size_)`.
If such a method doesn't exist, it should be added.